utils/frame/remote-externalities/src/lib.rs

```diff
@@ -320,72 +351,147 @@ where
 		Ok(keys)
 	}
 
-	/// Synonym of `rpc_get_pairs_unsafe` that uses paged queries to first get the keys, and then
+	/// Synonym of `getPairs` that uses paged queries to first get the keys, and then
 	/// map them to values one by one.
 	///
 	/// This can work with public nodes. But, expect it to be darn slow.
 	pub(crate) async fn rpc_get_pairs_paged(
 		&self,
 		prefix: StorageKey,
 		at: B::Hash,
+		pending_ext: &mut TestExternalities,
 	) -> Result<Vec<KeyValue>, &'static str> {
-		let keys = self.rpc_get_keys_paged(prefix, at).await?;
-		let keys_count = keys.len();
-		log::debug!(target: LOG_TARGET, "Querying a total of {} keys", keys.len());
-
-		let mut key_values: Vec<KeyValue> = vec![];
-		let client = self.as_online().rpc_client();
-		for chunk_keys in keys.chunks(BATCH_SIZE) {
-			let batch = chunk_keys
-				.iter()
-				.cloned()
-				.map(|key| ("state_getStorage", rpc_params![key, at]))
-				.collect::<Vec<_>>();
+		let keys = self.rpc_get_keys_paged(prefix.clone(), at).await?;
+		let client = self.as_online().transport.remote_client.clone().unwrap();
+		let thread_chunk_size = ((keys.len() + Self::threads() - 1) / Self::threads()).max(1);
```

Comment:
move this after the `keys.is_empty()` check or add a comment that is checked below i.e, that we can end-up with `thread_chunk_size == 0`

if you would force `Self::threads()` return `NonZeroUsize` you wouldn't need to add more clarification comments and the `max(1)` is unreachable.

Reply:
`max(1)` is needed to ensure that if e.g. `keys.len() == 16` and `threads == 20`, we don't allocate 1 key to each thread.

Reply:
done.